README.md

```diff
@@ -1,2 +1,114 @@
+
+## Architecture
+
+### Workload mTLS private key protection:
+<img src="diagrams/mTLS-arch.png">
+
+### Gateway private key protection:
+TODO: Add gateway arch diagram.
+## Prerequisites
+
+Prerequisites for using Istio mTLS private key protection with SGX:
+
+- Kubernetes cluster with one or more nodes with Intel® [SGX](https://software.intel.com/content/www/us/en/develop/topics/software-guard-extensions.html) supported hardware
+- [Intel® SGX device plugin for Kubernetes](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md)
+- Linux kernel version 5.11 or later on the host (in tree SGX driver)
+- [trusted-certificate-issuer](https://github.com/intel/trusted-certificate-issuer)
+- [Intel® SGX AESM daemon](https://github.com/intel/linux-sgx#install-the-intelr-sgx-psw)
+- [Intel® KMRA service](https://www.intel.com/content/www/us/en/developer/topic-technology/open/key-management-reference-application/overview.html)
+> NOTE: The KMRA service and AESM daemon is optional, needs to be set up only when remote attestaion required, which can be set through `NEED_QUOTE` flag in the chart.
+
+## Getting started
+
+This section covers how to install Istio mTLS and gateway private keys protection with SGX
+### Create signer
+
+```sh
+export CA_SIGNER_NAME=sgx-signer
+cat << EOF | kubectl create -f -
+apiVersion: tcs.intel.com/v1alpha1
+kind: TCSClusterIssuer
+metadata:
+    name: $CA_SIGNER_NAME
+spec:
+    secretName: ${CA_SIGNER_NAME}-secret
+    # If using quoteattestaion, set selfSign as false
+    # selfSign: false
+EOF
+```
+
+```sh
+# Get CA Cert and replace it in ./deployment/istio-configs/istio-hsm-config.yaml
+kubectl get secret -n tcs-issuer ${CA_SIGNER_NAME}-secret -o jsonpath='{.data.tls\.crt}' |base64 -d | sed -e 's;\(.*\);        \1;g'
+```
+### Protect the private keys of workloads with HSM
+1. Install Istio
+
+```sh
+istioctl install -f ./deployment/istio-configs/istio-hsm-config.yaml -y
+```
+
+> NOTE: You can also customize the `istio-hsm-config.yaml` according to your needs. If you want do the quote verification, you should set the `NEED_QUOTE` env as `true`. And if you are using the TCS v1alpha1 api, you can set the `RANDOM_NONCE` as `false`.
+
+
+
+2. Verifiy the Istio is ready
+
+By deault, `Istio` will be installed in the `istio-system` namespce
+
+```sh
+# Ensure that the pod is running state
+$ kubectl get po -n istio-system
+NAME                                    READY   STATUS    RESTARTS   AGE
+istio-ingressgateway-6cd77bf4bf-t4cwj   1/1     Running   0          70m
+istiod-6cf88b78dc-dthpw                 1/1     Running   0          70m
+```
+
+3. Create sleep and httpbin deployment:
+> NOTE: If you want use the sds-custom injection template, you need to set the annotations `inject.istio.io/templates` for both `sidecar` and `sgx`. And the ClusterRole is also required.
```

Comment:
Can we make the ClusterRole part of the SGX injection template?

Reply:
Currently we can't since since different workloads use their own `service account` name, and the `Cluster Role` should be bound with `service account`.